common/persistence/nosql/nosqlplugin/cassandra/workflow_utils.go

```diff
@@ -57,95 +59,79 @@ func executeCreateWorkflowBatchTransaction(
-			var columns []string
-			for k, v := range previous {
-				columns = append(columns, fmt.Sprintf("%s=%v", k, v))
-			}
-
-			if execution, ok := previous["execution"].(map[string]interface{}); ok {
-				// CreateWorkflowExecution failed because it already exists
-				executionInfo := parseWorkflowExecutionInfo(execution)
-				lastWriteVersion := common.EmptyVersion
-				if previous["workflow_last_write_version"] != nil {
-					lastWriteVersion = previous["workflow_last_write_version"].(int64)
-				}
-
-				msg := fmt.Sprintf("Workflow execution already running. WorkflowId: %v, RunId: %v, rangeID: %v, columns: (%v)",
-					executionInfo.WorkflowID, executionInfo.RunID, shardCondition.RangeID, strings.Join(columns, ","))
-
-				if currentWorkflowRequest.WriteMode == nosqlplugin.CurrentWorkflowWriteModeInsert {
-					return &nosqlplugin.WorkflowOperationConditionFailure{
-						WorkflowExecutionAlreadyExists: &nosqlplugin.WorkflowExecutionAlreadyExists{
-							OtherInfo:        msg,
-							CreateRequestID:  executionInfo.CreateRequestID,
-							RunID:            executionInfo.RunID,
-							State:            executionInfo.State,
-							CloseStatus:      executionInfo.CloseStatus,
-							LastWriteVersion: lastWriteVersion,
-						},
+			if currentWorkflowRequest.WriteMode == nosqlplugin.CurrentWorkflowWriteModeInsert {
+				currentExecutionAlreadyExists = true
+				actualExecution, _ = previous["execution"].(map[string]interface{})
+				if actualExecution != nil {
+					if previous["workflow_last_write_version"] != nil {
+						actualLastWriteVersion = previous["workflow_last_write_version"].(int64)
 					}
 				}
-				return &nosqlplugin.WorkflowOperationConditionFailure{
-					CurrentWorkflowConditionFailInfo: &msg,
+			} else if currentWorkflowRequest.WriteMode == nosqlplugin.CurrentWorkflowWriteModeUpdate {
+				if actualCurrRunID = previous["current_run_id"].(gocql.UUID).String(); requestConditionalRunID != "" && actualCurrRunID != requestConditionalRunID {
+					runIDMismatch = true
 				}
-			}
-
-			if prevRunID := previous["current_run_id"].(gocql.UUID).String(); requestConditionalRunID != "" && prevRunID != requestConditionalRunID {
-				// currentRunID on previous run has been changed, return to caller to handle
-				msg := fmt.Sprintf("Workflow execution creation condition failed by mismatch runID. WorkflowId: %v, Expected Current RunID: %v, Actual Current RunID: %v",
-					execution.WorkflowID, currentWorkflowRequest.Condition.GetCurrentRunID(), prevRunID)
-				return &nosqlplugin.WorkflowOperationConditionFailure{
-					CurrentWorkflowConditionFailInfo: &msg,
+				if currentWorkflowRequest.Condition != nil && currentWorkflowRequest.Condition.LastWriteVersion != nil {
+					ok := false
+					if actualLastWriteVersion, ok = previous["workflow_last_write_version"].(int64); ok && *currentWorkflowRequest.Condition.LastWriteVersion != actualLastWriteVersion {
+						lastWriteVersionMismatch = true
```

Comment:
same here. just return the err here instead of keeping a bool flag for it which is handled below. the more local the decision and action is the more readable/maintainable the code is